---
title: "Class 19"
author: Arturo Agramont
format: pdf
editor: visual
---

## Pertussis Cases by Year

-   **Q1.** With the help of the R \"addin\" package [**datapasta**](https://milesmcbain.github.io/datapasta/) assign the CDC pertussis case number data to a data frame called `cdc` and use **ggplot** to make a plot of cases numbers over time.

We will use datapasta to copy the table from the website as a data frame.

```{r}
cdc = data.frame(
                          Year = c(1922L,
                                   1923L,1924L,1925L,1926L,1927L,1928L,
                                   1929L,1930L,1931L,1932L,1933L,1934L,1935L,
                                   1936L,1937L,1938L,1939L,1940L,1941L,
                                   1942L,1943L,1944L,1945L,1946L,1947L,1948L,
                                   1949L,1950L,1951L,1952L,1953L,1954L,
                                   1955L,1956L,1957L,1958L,1959L,1960L,
                                   1961L,1962L,1963L,1964L,1965L,1966L,1967L,
                                   1968L,1969L,1970L,1971L,1972L,1973L,
                                   1974L,1975L,1976L,1977L,1978L,1979L,1980L,
                                   1981L,1982L,1983L,1984L,1985L,1986L,
                                   1987L,1988L,1989L,1990L,1991L,1992L,1993L,
                                   1994L,1995L,1996L,1997L,1998L,1999L,
                                   2000L,2001L,2002L,2003L,2004L,2005L,
                                   2006L,2007L,2008L,2009L,2010L,2011L,2012L,
                                   2013L,2014L,2015L,2016L,2017L,2018L,
                                   2019L,2020L,2021L),
  No..Reported.Pertussis.Cases = c(107473,
                                   164191,165418,152003,202210,181411,
                                   161799,197371,166914,172559,215343,179135,
                                   265269,180518,147237,214652,227319,103188,
                                   183866,222202,191383,191890,109873,
                                   133792,109860,156517,74715,69479,120718,
                                   68687,45030,37129,60886,62786,31732,28295,
                                   32148,40005,14809,11468,17749,17135,
                                   13005,6799,7717,9718,4810,3285,4249,
                                   3036,3287,1759,2402,1738,1010,2177,2063,
                                   1623,1730,1248,1895,2463,2276,3589,
                                   4195,2823,3450,4157,4570,2719,4083,6586,
                                   4617,5137,7796,6564,7405,7298,7867,
                                   7580,9771,11647,25827,25616,15632,10454,
                                   13278,16858,27550,18719,48277,28639,
                                   32971,20762,17972,18975,15609,18617,6124,
                                   2116)
)
```

We will now make a plot of the data frame.

```{r}
library(ggplot2)
ggplot(cdc) +
  aes(Year, No..Reported.Pertussis.Cases) +
  geom_point() +
  geom_line() +
  labs(x = "Year", y = "Number of cases")
```

# **2. A tale of two vaccines (wP & aP)**

-   **Q2.** Using the ggplot `geom_vline()` function add lines to your previous plot for the 1946 introduction of the wP vaccine and the 1996 switch to aP vaccine (see example in the hint below). What do you notice?

```{r}
ggplot(cdc) +
  aes(Year, No..Reported.Pertussis.Cases) +
  geom_point() +
  geom_line() +
  labs(x = "Year", y = "Number of cases") +
  geom_vline(xintercept = 1946,linetype = 2, color = "blue") +
  geom_vline(xintercept = 1996, linetype = 2, color = "red") +
  geom_text(aes(x=1950,y = 250000, label = "wP")) +
  geom_text(aes(x=2000,y = 250000, label = "aP")) 
```

-   **Q3.** Describe what happened after the introduction of the aP vaccine? Do you have a possible explanation for the observed trend?

After the introduction of the AP vaccine, the number of cases of Pertussis increased by year. A possible explanation for this is bacterial evolution, the bacteria evolving over time to become more resistant to immune system and effects of vaccines.

# **3. Exploring CMI-PB data**

We will use jsonlite package in order to read data from the CMI-PB API database. We will then explore this data

```{r}
library(jsonlite)
subject <- read_json("https://www.cmi-pb.org/api/subject", simplifyVector = TRUE) 

```

```{r}
head(subject, 3)
```

-   **Q4.** How many aP and wP infancy vaccinated subjects are in the dataset?

```{r}
table(subject$infancy_vac)
```

There are 47 aP vaccinated and 49 vaccinated subjects.

-   **Q5.** How many Male and Female subjects/patients are in the dataset?

```{r}
table(subject$biological_sex)

```

There are 66 females and 30 males

-   **Q6.** What is the breakdown of race and biological sex (e.g. number of Asian females, White males etc...)?

```{r}
table(subject$biological_sex, subject$race)
```

We will use lubridate to work with dates from this data

```{r}
library(lubridate)
today()
today() - ymd("2000-01-01")

```

```{r}
time_length( today() - ymd("2000-01-01"),  "years")

```

-   **Q7.** Using this approach determine (i) the average age of wP individuals, (ii) the average age of aP individuals; and (iii) are they significantly different?

```{r}

subject$age = today() - ymd(subject$year_of_birth)
```

```{r}
library(dplyr)

ap <- subject %>% filter(infancy_vac == "aP")

round( summary( time_length( ap$age, "years" ) ) )
```

```{r}
wp <- subject %>% filter(infancy_vac == "wP")
round( summary( time_length( wp$age, "years" ) ) )
```

```{r}
t.test(ap$age, wp$age)
```

These results are significantly different.

-   **Q8.** Determine the age of all individuals at time of boost?

```{r}
int <- ymd(subject$date_of_boost) - ymd(subject$year_of_birth)
age_at_boost <- time_length(int, "year")
head(age_at_boost)
```

-   **Q9.** With the help of a faceted boxplot or histogram (see below), do you think these two groups are significantly different?

```{r}
ggplot(subject) +
  aes(time_length(age, "year"),
      fill=as.factor(infancy_vac)) +
  geom_histogram(show.legend=FALSE) +
  facet_wrap(vars(infancy_vac), nrow=2) +
  xlab("Age in years")
```

The graph shows clear differences between the 2 groups, therefore they are most likely significantly different.

```{r}
specimen <- read_json("https://www.cmi-pb.org/api/specimen", simplifyVector = TRUE)
titer <- read_json("https://www.cmi-pb.org/api/ab_titer", simplifyVector = TRUE) 
```

Now we will join specimen and subject together

```{r}
meta <- inner_join(specimen, subject)
dim(meta)
head(meta)
```

-   **Q11.** Now using the same procedure join `meta` with `titer` data so we can further analyze this data in terms of time of visit aP/wP, male/female etc.

```{r}
abdata <- inner_join(titer, meta)
dim(abdata)
```

-   **Q12.** How many specimens (i.e. entries in `abdata`) do we have for each `isotype`?

```{r}
table(abdata$isotype)

```

-   **Q13.** What do you notice about the number of visit 8 specimens compared to other visits?

```{r}
table(abdata$visit)

```

The number of visit 8 specimens are much lower compared to the other visits

# **4. Examine IgG1 Ab titer levels**

We will use filter to exclude visit 8 entries

```{r}
ig1 <- abdata %>% filter(isotype == "IgG1", visit!=8)
head(ig1)
```

-   **Q14.** Complete the following code to make a summary boxplot of Ab titer levels (MFI) for all antigens:

We are going to make a v=boxplot of the Ab titer levels

```{r}
ggplot(ig1) +
  aes(MFI, antigen) +
  geom_boxplot() + 
  facet_wrap(vars(visit), nrow=2)
```

-   **Q15.** What antigens show differences in the level of IgG1 antibody titers recognizing them over time? Why these and not others?

FIM2/3 show a large difference in levels of IgG1 antibody titers.

```{r}
ggplot(ig1) +
  aes(MFI, antigen, col=infancy_vac ) +
  geom_boxplot(show.legend = FALSE) + 
  facet_wrap(vars(visit), nrow=2) +
  theme_bw()
```

```{r}
ggplot(ig1) +
  aes(MFI, antigen, col=infancy_vac ) +
  geom_boxplot(show.legend = FALSE) + 
  facet_wrap(vars(infancy_vac, visit), nrow=2)
```

-   **Q16.** Filter to pull out only two specific antigens for analysis and create a boxplot for each. You can chose any you like. Below I picked a \"control\" antigen (**\"Measles\"**, that is not in our vaccines) and a clear antigen of interest (**\"FIM2/3\"**, extra-cellular fimbriae proteins from *B. pertussis* that participate in substrate attachment).

```{r}
filter(ig1, antigen=="FIM2/3") %>%
  ggplot() +
  aes(MFI, col=infancy_vac) +
  geom_boxplot(show.legend = T) +
  facet_wrap(vars(visit)) +
  theme_bw()
```

```{r}
filter(ig1, antigen=="FHA") %>%
  ggplot() +
  aes(MFI, col=infancy_vac) +
  geom_boxplot(show.legend = T) +
  facet_wrap(vars(visit)) +
  theme_bw()
```

-   **Q17.** What do you notice about these two antigens time courses and the FIM2/3 data in particular?

Peak at 5, with a decline. FHA, and FIM2/3 both have an increase over time.

-   **Q18.** Do you see any clear difference in aP vs. wP responses?

wP increases more at first but then aP has a greater increase afterward.

# **5. Obtaining CMI-PB RNASeq data**

We are going to read RNA-seq data

```{r}
url <- "https://www.cmi-pb.org/api/v2/rnaseq?versioned_ensembl_gene_id=eq.ENSG00000211896.7"

rna <- read_json(url, simplifyVector = TRUE) 
```

We will join RNA data with meta data

```{r}
ssrna <- inner_join(rna, meta)
```

-   **Q19.** Make a plot of the time course of gene expression for IGHG1 gene (i.e. a plot of `visit` vs. `tpm`).

```{r}
ggplot(ssrna) +
  aes(visit, tpm, group=subject_id) +
  geom_point() +
  geom_line(alpha=0.2)
```

-   **Q20.**: What do you notice about the expression of this gene (i.e. when is it at it\'s maximum level)?

Visit 4 had the highest amount of expression. With a sharp decrease afterward.

-   **Q21.** Does this pattern in time match the trend of antibody titer data? If not, why not?

The titer data demonstrated that the highest expression occurred during visit 5, but in this instance visit 4 had the highest The cells expressed the gene earlier on than when the antibodies were at their greatest.
